specification/containerregistry/resource-manager/readme.md

```diff
@@ -36,6 +36,7 @@ These settings apply only when `--tag=package-2017-10` is specified on the comma
 ``` yaml $(tag) == 'package-2017-10'
 input-file:
 - Microsoft.ContainerRegistry/stable/2017-10-01/containerregistry.json
+- Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```

Comment:
For Python SDK, we currently have `registries` resource type in `2017-10-01` and `2017-03-01` versions https://github.com/Azure/azure-sdk-for-python/tree/master/azure-mgmt-containerregistry/azure/mgmt/containerregistry We want to keep supporting both versions.

For this new feature and version, should we move it to a new tag and generate the SDK to the third folder (e.g., `v2018_02_01-preview`) or we can put it in the same `v2017_10_01` folder? I am currently thinking the latter might make more sense since users don't have to switch version when creating a build after creating a registry. What is the concern with mixing preview and stable versions? What is the suggested approach here?

/cc @lmazuel @mnltejaswini @sajayantony

Reply:
Option 1: both under the same tag 2017-10
### Tag: package-2017-10

``` yaml $(tag) == 'package-2017-10'
input-file:
- Microsoft.ContainerRegistry/stable/2017-10-01/containerregistry.json
- Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json
```

Option 2: both under the same tag 2018-02-preview
### Tag: package-2018-02-preview

``` yaml $(tag) == 'package-2018-02-preview'
input-file:
- Microsoft.ContainerRegistry/stable/2017-10-01/containerregistry.json
- Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json
```


Option 3: each has it's own tag
### Tag: package-2018-02-preview

``` yaml $(tag) == 'package-2018-02-preview'
input-file:
- Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json
```

### Tag: package-2017-10

``` yaml $(tag) == 'package-2017-10'
input-file:
- Microsoft.ContainerRegistry/stable/2017-10-01/containerregistry.json
```

Reply:
Option 2 sounds right to me

Reply:
Option 2 seemed to be a breaking change though. The SDK had been referenced with the tag in it such as https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py#L11 or are you suggesting we should generate all folders in the SDK?

Reply:
I suggest the readme should have tags for both ` package-2018-02-preview` and ` package-2017-10`, where in the 2018 there are also references to the files you need from 2017.
[Example](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/readme.md#tag-package-2018-04)